File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            char lc = s.charAt(left);
            char rc = s.charAt(right);
            boolean lflag = false;
            boolean rflag = false;


            // non alpha numeric (post lower case method)
            if (lc < 48 || ( lc > 57 && lc < 96) ) left++;
            else lflag = true;
            if (rc < 48 || ( rc > 57 && rc < 96) ) right--;
            else rflag = true;

            if (lflag && rflag) {
                if (lc != rc) return false;
                left++;
                right--;
            }

        }

        return true;
    }
}

File: Data Structures & Algorithms/find-peak-element/submission-1.cs
public class Solution {
    public int FindPeakElement(int[] nums) {
        int highestPeakIndex = 0;
        if (nums.Length < 2) {
            return 0;
        }

        for (int i = 0; i < nums.Length; i++) {
            int right = i + 1;
            int left = i - 1;
            if (i > 0 && i < nums.Length - 1) {
                if (nums[i] > nums[right] && nums[i] > nums[left]) {
                    if (nums[i] > nums[highestPeakIndex]) {
                        highestPeakIndex = i;
                    }
                }
            }
            else if (i == 0) {
                if (nums[i] > nums[right]) {
                    if (nums[i] > nums[highestPeakIndex]) {
                        highestPeakIndex = i;
                    }
                }
            }
            else {
                if (nums[i] > nums[left]) {
                    if (nums[i] > nums[highestPeakIndex]) {
                        highestPeakIndex = i;
                    }
                }
            }
        }

        return highestPeakIndex;
    }
}